File: test/broadcast.cc
#include <rdc.h>
using namespace rdc;
int main(int argc, char *argv[]) {
    rdc::Init(argc, argv);
    std::string s;
    if (rdc::GetRank() == 0) s = "hello world";
    LOG_F(INFO, "@node[%d] before-broadcast: s=\"%s\"\n",
           rdc::GetRank(), s.c_str());
    // broadcast s from node 0 to all other nodes
    rdc::Broadcast(s, 0);
    LOG_F(INFO, "@node[%d] after-broadcast: s=\"%s\"\n",
           rdc::GetRank(), s.c_str());
    CHECK_F(s == "hello world");
    rdc::Finalize();
    return 0;
}

File: pytest/broadcast.py
#!/usr/bin/env
"""
demo python script of broadcast
"""
from __future__ import print_function
import os

rdc.init()
n = 3
rank = rdc.get_rank()
s = None
if rank == 0:
    s = {'hello world':100, 2:3}
print('@node[%d] before-broadcast: s=\"%s\"' % (rank, str(s)))
s = rdc.broadcast(s, 0)

print('@node[%d] after-broadcast: s=\"%s\"' % (rank, str(s)))
rdc.finalize()
